### ISSUE TITLE
Viewer attributen en pointer blijft staan maar is niet geselecteerd

### ISSUE THREAD

**Issue by gerardnienhuis** [IDgis/geoportaal-test#434]
Als je een laag kiest in de viewer, en daarna er op klikt, dan krijg je informatie over het betreffende object. Maar als je daarna de kaart beeldvullend maakt, en die laag uitzet, dan verdwijnt die laag niet altijd uit het kaartbeeld.

Komt bijna niet voor, dus vooralsnog niet als urgent ervaren. Maar het klopt niet, formeel.

bijvoorbeeld:

Naam van de service: Bodem
http://services.geodataoverijssel.nl/viewer?service=B41_bodem
Als je hier de laag Verontreinigingscontour grondwater aanzet, dan blijft deze staan.
Ook als deze later wordt uitgezet. 

Resultaat staat in onderstaande schermafdruk.

![viewer_bug_laag_blijft_aan](https://cloud.githubusercontent.com/assets/9005444/22608143/75609e1c-ea5b-11e6-83f5-f467c96832f9.jpg)

**Comment by gerardnienhuis** [comment]
Een vergelijkbaar verschijnsel komt voor, als je de informatie van een object opvraagt en dan de hele betreffende laag uitzet: die informatie komt op scherm maar blijft staan, ook als de laag niet meer geselecteerd is.

voorbeeld in schermafdruk van service Nutsvoorzienngen
http://services.geodataoverijssel.nl/viewer?service=B64_nutsvoorzieningen

Merk op: hier zie je element van laag Straalpaden, maar die laag is niet meer geselecteerd.
Het komt dus bij meer dan 1 laag voor.

![informatie_aanwezig_zonder_laag](https://cloud.githubusercontent.com/assets/9005444/22608483/5c9e35ae-ea5d-11e6-9bfc-d3c2aafd7428.jpg)

**Comment by SandroNeumann** [comment]
Manier om dit te produceren:

- laag aanklikken
- klik in kaart zodat marker verschijnt
- laag uitklikken
- laag aanklikken
- klik in kaart zodat marker verspringt
- laag uit -> marker wordt uitgezet ipv laag

**Comment by gerardnienhuis** [comment]
werkt nog niet goed, getest met:
http://acc-services.geodataoverijssel.nl/viewer?service=B41_bodem

stappen: 

1) aardkundige waarden aanklikken
2) attribuutinfo opvragen (klik midden in kaart): je ziet dan inhoud van het attribuut, gaat goed
3) dan aardkundige waarden vink weghalen: attribuutinfo blijft staan.
(is niet goed)

4) klik je daarna andere laag aan, dan verschijnt er niets meer op scherm. (is ook niet goed)
getest met acidofiele indicatiewaarde

**Comment by SandroNeumann** [comment]
De attribuutinfo wordt inderdaad niet weggehaald als je de laag uitvinkt. Gaan we oplossen.

Stap 4 werkt bij mij echter wel gewoon. Misschien dat je je cache moet legen?

**Comment by gerardnienhuis** [comment]
Helaas, stap 4 werkt bij mij niet. Sterker nog, als ik F5 indruk, dan komt er inderdaad een nieuw scherm, maar daarbij is alles uitgevinkt. Terwijl ik bij stap juist wel een andere laag had aangevinkt.

**Comment by gerardnienhuis** [comment]
zojuist getest: gedrag is ook bij andere services. Als je trouwens na de F5 weer de betreffende laag aanklikt, dan zie je die wel weer gewoon. Het is door meer onlogisch dan onwerkbaar.

**Comment by gerardnienhuis** [comment]
Ook getest: als ik gewoon begin met een service, bekijk in viewer en ik kies een laag die ik aanvink, dan zie ik de resultaten in kaart. Maar dan is altijd na F5 de betreffende vink en dus ook de kaart weer weg.

**Comment by SandroNeumann** [comment]
Hoezo zou je verwachten dat je na F5 nog de vinks aan hebt staan? Dan laad je de pagina opnieuw, dus ben je altijd al je acties kwijt.

**Comment by gerardnienhuis** [comment]
helemaal gelijk in, dus na F5 leeg scherm is logisch, maar hoe leeg ik dan mijn cache, zoals je mij vraagt te doen? Doe ik tot nu altijd met F5 of Ctrl F5

**Comment by SandroNeumann** [comment]
Ik weet dat het in Chrome shift + F5 is.

**Comment by gerardnienhuis** [comment]
shift F5 in Chrome maakt scherm leeg maar lost dus probleem niet op. 

Nogmaals getest met actuele versies van verschillende browsers: IE(11), FF, Chrome en Opera.
Dit geeft allemaal zelfde probleem.

Getest met: http://acc-wmsonly-services.geodataoverijssel.nl/viewer?service=B41_bodem_alleen_wms

stappen:

1) vink geomorf. kaart aan: kaart verschijnt (correct)
2) klik een gebied aan, in de kaart: dan zie je attribuutbeschrijving onder in scherm (correct)
3) vink geomorf. kaart uit:  kaart verdwijnt (correct) maar attribuutbeschrijving blijft bestaan (niet correct)
4) vink bodemkaart aan: kaart verschijnt (correct) maar attribuutbeschrijving van geomorf. kaart staat er nog steeds (niet correct)

resultaat is te zien in onderstaande figuur:

![viewer_verkeerde_attributen](https://user-images.githubusercontent.com/9005444/32363200-6065f310-c06e-11e7-98a5-7944180b3143.jpg)

**Comment by SandroNeumann** [comment]
Hier kan ik stap 4 wel reproduceren, maar eerder in dit issue zei je dat bij opnieuw aanvinken van een laag die laag niet getoond wordt.

Zou het kunnen dat je dat laatste alleen in IE hebt? Dan heeft het zeer waarschijnlijk te maken met #521.

**Comment by gerardnienhuis** [comment]
Correct, er lijken minstens 3 problemen te zijn. Die ook nog eens browserafhankelijk zijn. Ik test nu thuis met IE-11 en FF, Chrome en Opera. Maar gisteren testte ik met IE (x) in het provinciehuis. Met andere effecten. Als ik nu de stappen herhaal, dan krijg ik wel de nieuwe laag te zien, zie ook figuur hiervoor. 

Maar als ik datzelfde doe in het provinciehuis, dan verschijnt die nieuwe laag niet eens meer, lijkt het. Hiervan maak ik een ander issue.

Dit issue gaat namelijk over het feit dat de attributen onterecht blijven staan, ook als een andere kaart is aangevinkt.

**Comment by SandroNeumann** [comment]
Ik heb net ook in IE getest en daar verschijnt de laag niet opnieuw. Heeft te maken met de marker die niet verschijnt.

Volgens mij zijn dit de issues:
- attribuutinfo weghalen als je de laag uit klikt
- marker in IE
- laag bug in IE

**Comment by gerardnienhuis** [comment]
zojuist aangemaakt:  https://github.com/IDgis/geoportaal-test/issues/522

**Comment by SandroNeumann** [comment]
Ik heb een nieuwe versie van de viewer op acceptatie uitgerold waarmee de problemen opgelost zouden moeten zijn. Ook die van #522.

**Comment by gerardnienhuis** [comment]
Zojuist getest in provinciehuis Overijssel, met IE-11.

getest met: http://acc-services.geodataoverijssel.nl/viewer?service=B41_bodem 

De attributen van de eerst geselecteerde kaart blijven nog steeds in beeld, en geeft dus onjuiste informatie op scherm.

stappen:

1) vink aardkundige waarden aan: kaart komt op scherm, is juist
2) klik ergens op scherm: attributen komen onder in beeld: is juist
3) vink aardkundige waarden uit, attributen blijven onder in beeld: is onjuist
4) vink acidofiele indicatiewaare aan: kaart komt op scherm (juist) maar attributen staan nog steeds onderaan scherm (dus van de aardkundige waarden, is onjuist).

zie figuur, resultaat van stap 4:

![fout_attributen_oude_laag](https://user-images.githubusercontent.com/9005444/32440286-cd843eee-c2f2-11e7-9ef2-9e4cce5aeb50.jpg)

**Comment by gerardnienhuis** [comment]
merk op: het oorspronkelijke probleem mbt viewerlaag die niet weggaat, dat is al opgelost, maar het probleem met de attributen die blijven staan niet. Ter overweging: apart issue van maken?

**Comment by SandroNeumann** [comment]
Attributen blijven alleen staan in IE. In andere browsers worden de attributen wel weggehaald.

Dit zoeken we uit.

**Comment by gerardnienhuis** [comment]
Goed te weten dat andere browsers dit probleem niet hebben. Is minder urgent hierdoor. Issue van milestone 1.2 naar 1.3 verplaatst. 

Merk op: alleen bij IE blijven dus pointer en attributen staan, als er geklikt wordt op ge kaart. Klik je ergens anders, bij een eventuele nieuwe kaart, dan is dat probleem al opgelost. Dan verschijnt correct de pointer op de nieuwe plaats en komt ook de juiste attribuutinformtie in beeld.

Er is dus al een workaround beschikbaar. Het hoort niet zo te werken, dus daarom is het wel een bug, maar met minder hoge prioriteit.

**Comment by gerardnienhuis** [comment]
Minder hoge prio (komt bijna nooit voor, en is niet zeer relevant voor gebruikers): verplaats naar milestone 1.5
